queue_services/payment-reconciliations/src/reconciliations/payment_reconciliations.py

```diff
@@ -480,6 +481,8 @@ def _process_failed_payments(row):
         invoice: InvoiceModel = InvoiceModel.find_by_id(identifier=inv_reference.invoice_id)
         invoice.invoice_status_code = InvoiceStatus.SETTLEMENT_SCHEDULED.value
         invoice.paid = 0
+        # Add invoice to non sufficient funds table
+        NonSufficientFundsService.save_non_sufficient_funds(invoice_id=invoice.id, description='NSF')
```

Comment:
I think you're commiting here each time in the loop (via .save)?
The code used to commit in `_create_nsf_invoice`, should be ok